CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs

```diff
@@ -1,284 +1,100 @@
-        {
-            if (hero != null && hero.HeroState != CharacterStates.Dead && _newCulture != hero.Culture)
+            //count amount of assimilated settlements of region
+            float assimilatedCount = 0;
+            float villagesCount = _settlement.BoundVillages.Count;
+            for (int i = 0; i < villagesCount; i++)
             {
-                if (hero.IsVampireNotable())
-                {
-                    KillCharacterAction.ApplyByMurder(hero, null, false);
-                }
-                else if (hero.IsEmpireNotable())
-                {
-                    if (_newCulture.Name.Contains("Vampire"))
-                    {
-                        hero.TurnIntoVampire();
-                        if (!hero.CharacterObject.IsFemale)
-                        {
-                            if (hero.CurrentSettlement == Hero.MainHero.CurrentSettlement)
-                            {
-                                this.ReplaceImageIdentifier(hero.CharacterObject);
-                            }
-                        }
-                        return;
-                    }
-                }
-                if (hero.HeroState == CharacterStates.Dead)
-                {
-                    CreateNewNotable(hero);
-                }
+                var village = _settlement.BoundVillages[i].Settlement;
+                if (village.Culture == _settlement.Owner.Culture) assimilatedCount++;
             }
+            if (_settlement.Culture == _settlement.Owner.Culture) assimilatedCount++;
+            _assimilationProgress = assimilatedCount / (villagesCount + 1f);
         }
 
-        private void CreateNewNotable(Hero victim)
+        private void TryToAssimilateSettlement(Settlement settlement)
         {
-            Hero newNotable = HeroCreator.CreateRelativeNotableHero(victim);
-            EnterSettlementAction.ApplyForCharacterOnly(newNotable, newNotable.CurrentSettlement);
-            if (newNotable.CurrentSettlement.IsEmpireSettlement())
-            {
-                newNotable.TurnIntoHuman();
-            }
-            else if (newNotable.CurrentSettlement.IsVampireSettlement())
-            {
-                newNotable.TurnIntoVampire();
-            }
-            foreach (Hero otherHero in Hero.AllAliveHeroes)
-            {
-                int relation = victim.GetRelation(otherHero);
-                if (relation != 0)
-                {
-                    newNotable.SetPersonalRelation(otherHero, relation);
-                }
-            }
-            if (victim.Issue != null)
+            var outriders = settlement.Notables.Where(n => n.Culture != _settlement.Owner.Culture);
+            if (outriders.Count() > 0)
             {
-                Campaign.Current.IssueManager.ChangeIssueOwner(victim.Issue, newNotable);
+                var outrider = outriders.First();
+                outrider.Culture = _settlement.Owner.Culture;
```

Comment:
I am not sure that setting the notable's culture directly changes the troops recruitable by them. Have you tested? If yes, that is very good news! I thought we would have to create new notables based on the xml settings in culture for it to work.,

Reply:
I tested it a few months ago and it worked like a swiss watch

Reply:
Saw your picture as well on discord. Fine then, no more issues.